crates/breez-sdk/common/src/sync/model.rs

```diff
@@ -0,0 +1,506 @@
+use std::{collections::HashMap, fmt::Display};
+
+use bitcoin::hashes::{Hash, sha256};
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+const CURRENT_SCHEMA_VERSION: Version = Version::new(0, 2, 6);
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+pub struct RecordId {
+    pub r#type: String,
+    pub data_id: String,
+}
+
+impl RecordId {
+    pub fn new(r#type: impl Into<String>, data_id: impl Into<String>) -> Self {
+        RecordId {
+            r#type: r#type.into(),
+            data_id: data_id.into(),
+        }
+    }
+}
+
+impl Display for RecordId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{:x}",
+            sha256::Hash::hash(format!("{}:{}", self.r#type, self.data_id).as_bytes())
+        )
+    }
+}
```

Comment:
Nit: Should we implement a `hash` method? IMO, we should leave Display for readable formats